test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/RepositoryCountersignatureTests.cs

```diff
@@ -0,0 +1,400 @@
+
+                // Load primary signature
+                var primarySignature = PrimarySignature.Load(test.PrimarySignedCms.Encode());
+
+                // Validate countersignature
+                var countersignature = RepositoryCountersignature.GetRepositoryCountersignature(primarySignature);
+
+                Assert.Equal(SignatureType.Repository, countersignature.Type);
+            }
+        }
+
+        [Fact]
+        public void GetRepositoryCounterSignature_WithValidRepositoryCountersignature_ReturnsRepositoryCountersignature()
+        {
+            using (var test = new LoadTest(_fixture))
+            {
+                test.CreateValidAuthorPrimarySignature();
+
+                // Generate valid countersignature
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateCommitmentTypeIndication(SignatureType.Repository));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                     AttributeUtility.CreateNuGetV3ServiceIndexUrl(new Uri("https://api.nuget.org/v3/index.json")));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateNuGetPackageOwners(new List<string> { "microsoft", "nuget" }));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    new Pkcs9SigningTime());
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateSigningCertificateV2(test.Certificate, HashAlgorithmName.SHA256));
+
+                // Create counter signature
+                test.PrimarySignedCms.SignerInfos[0].ComputeCounterSignature(test.CounterCmsSigner);
+
+                // Load primary signature
+                var primarySignature = PrimarySignature.Load(test.PrimarySignedCms.Encode());
+
+                // Validate countersignature
+                var countersignature = RepositoryCountersignature.GetRepositoryCountersignature(primarySignature);
+
+                Assert.Equal(SignatureType.Repository, countersignature.Type);
+            }
+        }
+
+        private sealed class LoadTest : IDisposable
+        {
+            private bool _isDisposed;
+
+            internal X509Certificate2 Certificate { get; }
+            internal CmsSigner CounterCmsSigner { get; }
+
+            internal SignedCms PrimarySignedCms { get; }
+
+            internal LoadTest(CertificatesFixture fixture)
+            {
+                Certificate = fixture.GetDefaultCertificate();
+
+                var content = new SignatureContent(
+                    SigningSpecifications.V1,
+                    HashAlgorithmName.SHA256,
+                    hashValue: "hash");
+                var contentInfo = new ContentInfo(content.GetBytes());
+
+                PrimarySignedCms = new SignedCms(contentInfo);
+                CounterCmsSigner = new CmsSigner(Certificate);
```

Comment:
In this test class I think it's more appropriate to use:

    SigningUtility.CreateSignedAttributes(...)
    SigningUtility.CreateCmsSigner(...)

Reply:
I avoided doing this because `SignPackageRequest` takes ownership of the certificate, so instead of having multiple copies of the certificate I preferred doing the signing and countersigning manually to make it more readable.